### ISSUE TITLE
Bots under sub-categories of a site are being removed

### ISSUE THREAD

**Issue by autumnfound** [EclipseFdn/eclipsefdn-github-sync#169]
Currently, bot checks are strict in terms of looking up the bot name registered for a site ("github.com", "gitlab.com"). This means that any bots that are registered under a sub-category/grouping for the site (e.g. "github.com-openshift-ci-robot") are not registered as valid bots for a given site. There should be a way to read multiple bots for a given site and not removing them.

//cc @mbarbero @fredg02

**Comment by autumnfound** [comment]
A way we could address this issue would be to match on the passed site name with a pattern like `$sitename.*`, and return lists with multiple entries when we are looking up bots for sites/users. Are there any concerns with that solution?

**Comment by fredg02** [comment]
How is this handled in the ECA check for cases like "github.com-dependabot"? Using `$sitename.*` sounds like a pragmatic approach to me.

**Comment by autumnfound** [comment]
> How is this handled in the ECA check for cases like "github.com-dependabot"? Using `$sitename.*` sounds like a pragmatic approach to me.

It got pushed back to a PR as there was an NPE. Its my last PR to roll into master before I poke about testing to roll up to production. https://github.com/EclipseFdn/git-eca-rest-api/pull/50/files?diff=unified&w=1#diff-de47b295e8c83f448493870312e9ce5e9eb7d78a36e8795f8040da0098ade8e8R317

TLDR: It checks all emails for a bot given a project (instead of username), rather than just the given site bots

**Comment by fredg02** [comment]
If @mbarbero does not have any concerns, then let's go with the pragmatic approach. The Che project is still affected.

**Comment by mbarbero** [comment]
no concerns.

**Comment by autumnfound** [comment]
I've created a patch for this! Looking to merge in a different change set that has test+lint fixes as this will fail otherwise, but it shouldn't be too long

**Comment by autumnfound** [comment]
This was merged in, and has run against the production dataset @fredg02 can we validate this issue is resolved? If not, can we add the openshift bots to the Che project and test this with the 12pm EST run of the sync?

**Comment by fredg02** [comment]
I've re-added the openshift bots.

**Comment by fredg02** [comment]
Openshift bots did not get removed this time. So I call this a success.
Fixed by  #175.
